test/controllers/contactPreference/AddEmailAddressControllerSpec.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.contactPreference
+
+import common.SessionKeys
+import controllers.ControllerBaseSpec
+import play.api.http.Status.{NOT_FOUND, OK}
+import play.api.test.Helpers._
+import views.html.contactPreference.AddEmailAddressView
+
+class AddEmailAddressControllerSpec extends ControllerBaseSpec {
```

Comment:
Need to add a scenario for an unauthorised user and a user that currently has a digital preference. I'd also suggest structuring the tests slightly differently:
```
"AddEmailAddressController .show is called" when {

  "user is authorised" when {

    "user has paper preference" when {

      "feature switch is on" should {

      }

      "feature switch is off" should {

      }
    }
  
    "user has a digital preference" should {

    }
  }

  "user is unauthorised" should {
   
  }
}
```
as this reflects the order the logic applies (including the predicates), so its sometimes useful to lay out the tests in the same fashion